### PR TITLE
Checkin the first photon fast simulation module based on GAN

### PR DESCRIPTION
**Pull request by Healthborn**
Checkin include:
- change the name of the directory to avoid conflict;
- the CMakeLists.txt
- the code for the module
- the code for the tool

### CODE REVIEW

**Comment by lgarren** [comment]
@Healthborn @knoepfel  I'm taking a first look at this.  
First, my apologies, but please remove find_ups_product( eigen ) from the top level CMakeLists.txt file.  The ups eigen package is not used by larsimdnn.   Second, please update the larsim version to v09_08_00 in ups/product_deps.  Third, extensions matter.  If a file is meant to be included in a library, it needs the .cxx extension.  The .cc extension is used only for plugins.

File: larsimdnn/PhotonSimulation/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                       nug4_ParticleNavigation
+                       lardataobj_Simulation
+                       larevt_Filters
+                       lardataobj_RawData
+                       larcorealg_Geometry
+                       larcore_Geometry_Geometry_service
+                       ${ART_ROOT_IO_TFILE_SUPPORT} ROOT::Core
+                       ${ART_ROOT_IO_TFILESERVICE_SERVICE}
+                       ${MF_MESSAGELOGGER}
+                       ROOT::GenVector
```
Comment (lgarren) [inline larsimdnn/PhotonSimulation/CMakeLists.txt:34]: Please remove the SERVICE_LIBRARIES block.  There are no plugins in this directory.

File: larsimdnn/PhotonSimulation/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                        ${ART_ROOT_IO_TFILESERVICE_SERVICE}
+                        ${ART_FRAMEWORK_SERVICES_REGISTRY}
+                        ${ART_ROOT_IO_TFILE_SUPPORT}
+                        ROOT::Core
+                        ${MF_MESSAGELOGGER}
+                        nurandom_RandomUtils_NuRandomService_service
+                        art_Persistency_Provenance
+                        ${CLHEP}
+                        ROOT::GenVector
+                        ROOT::Gpad
```
Comment (lgarren) [inline larsimdnn/PhotonSimulation/CMakeLists.txt:53]: Please remove the MODULE_LIBRARIES block.  There are no plugins in this directory.

**Review by lgarren — CHANGES_REQUESTED** [review]

File: larsimdnn/PhotonSimulation/CMakeLists.txt (hunk 2 above)
Comment (lgarren) [reply larsimdnn/PhotonSimulation/CMakeLists.txt:53]: Oh, crud.  Sorry.  There is a module.

**Comment by lgarren** [comment]
@Healthborn @knoepfel 
After local modifications, I have a successful build which creates the following libraries.   I'm attaching a patch file.  The patch does not include renaming TFLoader.cc to TFLoader.cxx.  Kyle will be reviewing the code.
liblarsimdnn_PhotonSimulation_PDFastSimGAN_module.so
liblarsimdnn_PhotonSimulation_TFLoaderTools.so
liblarsimdnn_PhotonSimulation_TFLoaderTools_TFLoaderGAN_tool.so

Have to rename the file to attach.
[larsimdnn.patch.log](https://github.com/LArSoft/larsimdnn/files/5906714/larsimdnn.patch.log)

**Comment by Healthborn** [comment]
thanks a lot @lgarren, a bit busy this week. I will also exam the code.

**Comment by lgarren** [comment]
@Healthborn any progress?

**Pull request closed without merging**